src/include/sof/lib/clk.h

```diff
@@ -30,7 +30,11 @@ struct clock_notify_data {
 struct freq_table {
 	uint32_t freq;
 	uint32_t ticks_per_msec;
-	uint32_t enc;
+	uint32_t enc; /**< value to set clock */
+	uint32_t enc_status_mask; /**< mask to check clock status
+				   * value 0 means that there is no need for
+				   * checking clock status after setting it
+				   */
```

Comment:
@lgirdwood I didn't add them, cos I didn't know what we want to do with them.
I can change it to callbacks. Do you want something more to do with clocks - like f.e. splitting process into requesting/commiting instead of having blocking wait in set?

Reply:
@lgirdwood IMHO it's not worth extending api if we don't have case for it atm in current codebase.

Reply:
@jajanusz I want it to be more like Zephyr/Posix for easier integration so get/set is best (and is widely used).

Reply:
@lgirdwood I got back to it and I'm not sure what you want to achieve here. By making generic get/set do you mean removing enc&enc_status and hiding freq tables, so we will end up with another table mapping clocks to register masks OR just replacing these 2 value fields with 2 function pointers - get/set ?

Also `callback` sounds like a wrong word here, I guess you just meant replacing values checks with platform-specific function calls, cos I don't think there is suitable place for callbacks (f.e. some function passed to set that is executed in-some-time when clock is set).

Reply:
> @lgirdwood I got back to it and I'm not sure what you want to achieve here. By making generic get/set do you mean removing enc&enc_status and hiding freq tables, so we will end up with another table mapping clocks to register masks OR just replacing these 2 value fields with 2 function pointers - get/set ?
> 
I mean make the clock core generic and push the masks/values into the platform code using the get()/set(). Generally in the same way as Linux/Zephyr. Our clock code is too tightly tied to the current Intel arch atm which is probably causing @dbaluta pain today and maybe cause us pain tomorrow.

> Also `callback` sounds like a wrong word here, I guess you just meant replacing values checks with platform-specific function calls, cos I don't think there is suitable place for callbacks (f.e. some function passed to set that is executed in-some-time when clock is set).

Yes sorry, that's what I was meaning.